core/domain/user_services.py

```diff
@@ -1575,9 +1583,33 @@ def add_user_role(user_id: str, role: str) -> None:
         user_id, feconf.ROLE_ACTION_ADD, role=role,
         username=user_settings.username)
 
+    if role in [feconf.ROLE_ID_BLOG_ADMIN, feconf.ROLE_ID_BLOG_POST_EDITOR]:
+        create_aggregated_author_blog_post_stats_models(user_id)
+
     save_user_settings(user_settings)
 
 
+def create_aggregated_author_blog_post_stats_models(
```

Comment:
Should have something like "if does not exist already" in the name since you aren't doing anything if the model exists already.

Reply:
Done.